diff --git a/Makefile b/Makefile
index 72af3b0..7eb77a8 100644
--- a/Makefile
+++ b/Makefile
@@ -59,16 +59,19 @@ rtptrans_OBJS	= hpt.o host2ip.o notify.o multimer.o      rtptrans.o
 HAVE_SRCS = \
 	have-err.c		\
 	have-gethostbyname.c	\
+	have-progname.c		\
 	have-strtonum.c		\
 	have-msgcontrol.c
 
 COMPAT_SRCS = \
-	compat_err.c		\
-	compat_strtonum.c
+	compat-err.c		\
+	compat-progname.c	\
+	compat-strtonum.c
 
 COMPAT_OBJS = \
-	compat_err.o		\
-	compat_strtonum.o
+	compat-err.o		\
+	compat-progname.o	\
+	compat-strtonum.o
 
 OBJS =	$(rtpdump_OBJS) $(rtpplay_OBJS) $(rtpsend_OBJS) $(rtptrans_OBJS)
 OBJS +=	$(COMPAT_OBJS)
diff --git a/Makefile.depend b/Makefile.depend
index d099c47..24f2d1c 100644
--- a/Makefile.depend
+++ b/Makefile.depend
@@ -4,7 +4,12 @@ hsearch.o: hsearch.c sysdep.h hsearch.h
 multimer.o: multimer.c multimer.h notify.h sysdep.h
 notify.o: notify.c sysdep.h notify.h multimer.h ansi.h
 rd.o: rd.c rtpdump.h sysdep.h
+
 rtpdump.o: rtpdump.c rtp.h sysdep.h vat.h rtpdump.h ansi.h
 rtpplay.o: rtpplay.c sysdep.h notify.h rtp.h rtpdump.h multimer.h ansi.h
 rtpsend.o: rtpsend.c notify.h rtp.h sysdep.h multimer.h ansi.h
 rtptrans.o: rtptrans.c rtp.h sysdep.h rtpdump.h ansi.h notify.h multimer.h vat.h
+
+compat-err.o: compat-err.c
+compat-progname.o: compat-progname.c
+compat-strtonum.o: compat-strtonum.c
diff --git a/configure b/configure
index 302e0fd..a63afc8 100755
--- a/configure
+++ b/configure
@@ -40,23 +40,27 @@ LDFLAGS=
 LDADD=
 
 #HAVE_ENDIAN=
-#HAVE_ERR=
-HAVE_LNSL=
-HAVE_MSGCONTROL=
+HAVE_ERR=
 #HAVE_NTOHL=
 #HAVE_PATH_MAX=
-#HAVE_PROGNAME=
+HAVE_PROGNAME=
 #HAVE_REALLOCARRAY=
 #HAVE_RECALLOCARRAY=
-HAVE_LSOCKET=
 #HAVE_STRCASESTR=
 #HAVE_STRINGLIST=
 #HAVE_STRLCAT=
 #HAVE_STRLCPY=
 #HAVE_STRNDUP=
-#HAVE_STRTONUM=
+HAVE_STRTONUM=
 #HAVE_SYS_ENDIAN=
 
+# structures
+HAVE_MSGCONTROL=
+
+# extra libs
+HAVE_LNSL=
+HAVE_LSOCKET=
+
 INSTALL="install"
 PREFIX="/usr/local"
 BINDIR=
@@ -136,24 +140,28 @@ runtest() {
 	return 1
 }
 
-# --- library functions ---
+# functions
 #runtest be32toh		ENDIAN		|| true
 #runtest be32toh		SYS_ENDIAN	-DSYS_ENDIAN || true
-#runtest err		ERR		|| true
-runtest gethostbyname   LNSL		-lnsl	|| true
-runtest msgcontrol	MSGCONTROL	|| true
+runtest err		ERR		|| true
 #runtest ntohl		NTOHL		|| true
 #runtest PATH_MAX	PATH_MAX	|| true
-#runtest progname	PROGNAME	|| true
+runtest progname	PROGNAME	|| true
 #runtest reallocarray	REALLOCARRAY	|| true
 #runtest recallocarray	RECALLOCARRAY	|| true
-runtest socket		LSOCKET        -lsocket	|| true
 #runtest strcasestr	STRCASESTR	|| true
 #runtest stringlist	STRINGLIST	|| true
 #runtest strlcat		STRLCAT		|| true
 #runtest strlcpy		STRLCPY		|| true
 #runtest strndup		STRNDUP		|| true
-#runtest strtonum	STRTONUM	|| true
+runtest strtonum	STRTONUM	|| true
+
+# structures
+runtest msgcontrol	MSGCONTROL	|| true
+
+# extra libraries
+runtest gethostbyname	LNSL	-lnsl	|| true
+runtest socket		LSOCKET	-lsocket|| true
 
 #if [ ${HAVE_ENDIAN} -eq 0 -a \
      #${HAVE_SYS_ENDIAN} -eq 0 -a \
@@ -169,7 +177,7 @@ exec > config.h
 
 cat << __HEREDOC__
 #ifdef __cplusplus
-#error "Do not use C++.  See the INSTALL file."
+#error "Do not use C++."
 #endif
 
 #if !defined(__GNUC__) || (__GNUC__ < 4)
@@ -197,12 +205,13 @@ __HEREDOC__
 cat << __HEREDOC__
 /*
 #define HAVE_ENDIAN ${HAVE_ENDIAN}
-#define HAVE_ERR ${HAVE_ERR}
 */
-#define HAVE_MSGCONTROL ${HAVE_MSGCONTROL}
+#define HAVE_ERR ${HAVE_ERR}
 /*
 #define HAVE_NTOHL ${HAVE_NTOHL}
+*/
 #define HAVE_PROGNAME ${HAVE_PROGNAME}
+/*
 #define HAVE_REALLOCARRAY ${HAVE_REALLOCARRAY}
 #define HAVE_RECALLOCARRAY ${HAVE_RECALLOCARRAY}
 #define HAVE_STRCASESTR ${HAVE_STRCASESTR}
@@ -210,24 +219,28 @@ cat << __HEREDOC__
 #define HAVE_STRLCAT ${HAVE_STRLCAT}
 #define HAVE_STRLCPY ${HAVE_STRLCPY}
 #define HAVE_STRNDUP ${HAVE_STRNDUP}
+*/
 #define HAVE_STRTONUM ${HAVE_STRTONUM}
+/*
 #define HAVE_SYS_ENDIAN ${HAVE_SYS_ENDIAN}
 */
 
+#define HAVE_MSGCONTROL ${HAVE_MSGCONTROL}
+
 __HEREDOC__
 
-#if [ ${HAVE_ERR} -eq 0 ]; then
-	#echo "extern	void	  err(int, const char *, ...);"
-	#echo "extern	void	  errx(int, const char *, ...);"
-	#echo "extern	void	  warn(const char *, ...);"
-	#echo "extern	void	  warnx(const char *, ...);"
-#fi
+if [ ${HAVE_ERR} -eq 0 ]; then
+	echo "extern	void	  err(int, const char *, ...);"
+	echo "extern	void	  errx(int, const char *, ...);"
+	echo "extern	void	  warn(const char *, ...);"
+	echo "extern	void	  warnx(const char *, ...);"
+fi
+
+if [ ${HAVE_PROGNAME} -eq 0 ]; then
+	echo "extern const char*	getprogname(void);"
+	echo "extern void		setprogname(const char *);"
+fi
 
-#if [ ${HAVE_PROGNAME} -eq 0 ]; then
-	#echo "extern 	const char *getprogname(void);"
-	#echo "extern	void	  setprogname(const char *);"
-#fi
-#
 #[ ${HAVE_REALLOCARRAY} -eq 0 ] && \
 	#echo "extern	void	 *reallocarray(void *, size_t, size_t);"
 #
@@ -245,8 +258,8 @@ __HEREDOC__
 #
 #[ ${HAVE_STRNDUP} -eq 0 ] && \
 	#echo "extern	char	 *strndup(const char *, size_t);"
-#
-#[ ${HAVE_STRTONUM} -eq 0 ] && \
+
+[ ${HAVE_STRTONUM} -eq 0 ] && \
 	#echo "extern	long long strtonum(const char *, long long, long long, const char **);"
 
 echo "config.h: written" 1>&2
diff --git a/configure.local.example b/configure.local.example
index c5147ac..9bbcc4e 100644
--- a/configure.local.example
+++ b/configure.local.example
@@ -70,7 +70,6 @@ CFLAGS="-g -W -Wall -Wstrict-prototypes -Wno-unused-parameter -Wwrite-strings"
 
 HAVE_ENDIAN=0
 HAVE_ERR=0
-HAVE_MSGCONTROL=0
 HAVE_NTOHL=0
 HAVE_PATH_MAX=0
 HAVE_PROGNAME=0
@@ -84,5 +83,7 @@ HAVE_STRPTIME=0
 HAVE_STRTONUM=0
 HAVE_SYS_ENDIAN=0
 
+HAVE_MSGCONTROL=0
+
 HAVE_LSOCKET=0
 HAVE_LNSL=0
